components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -4027,8 +4027,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     }
 
     async getOnboardingState(ctx: TraceContext): Promise<GitpodServer.OnboardingState> {
-        this.checkAndBlockUser("getOnboardingState");
```

Comment:
This call needs to be available to unauth'd users since we use it's results to determine what to show on the login page. I wanted to drop a comment here in case we need to worry about that change though. Will our rate limit be sufficient protection around this method, or do we need to look at caching the results of this call in Redis?

Reply:
I was getting back a 401 until I removed it, and it looks like we do enforce a user in that method [here](https://github.com/gitpod-io/gitpod/blob/a3693d3636019d4496d8f0d7700c729c2ffe89bf/components/server/src/workspace/gitpod-server-impl.ts#L633).

Reply:
@easyCZ, `checkAndBlockUser` does the basic authZ, it tests if the session is authenticated. 

```
        // Generally, a user session is required.
        if (!this.user) {
            throw new ResponseError(ErrorCodes.NOT_AUTHENTICATED, "User is not authenticated. Please login.");
        }
```
The client connection can be authenticated or not, which also doesn't change during the lifetime of the connection. There are only a few exceptions to the RPC methods which are allowed to work unauthenticated. `getOnboardingState` didn't need to work unauthenticated yet, as its result was only required during the onboarding session. 

That change here makes sense, as we want to show the state for every visitor.